common/src/commonMain/kotlin/com/trikot/sample/factories/ViewModelControllerFactoryImpl.kt

```diff
@@ -0,0 +1,12 @@
+package com.trikot.sample.factories
+
+import com.mirego.trikot.kword.I18N
+import com.trikot.sample.domain.impl.FetchQuoteUseCaseImpl
+import com.trikot.sample.repositories.impl.QuoteRepositoryImpl
+import com.trikot.sample.viewmodels.home.HomeViewModelController
+
+class ViewModelControllerFactoryImpl(private val i18N: I18N) :
+    ViewModelControllerFactory {
+    override val homeViewModelController: HomeViewModelController get() =
```

Comment:
I don't like factory method to be a property. I think this is not clear enough that a new object will be create each time. I prefer a method with the prefix create and avoiding to repeat ViewModelController every time. So for the example here: 
`override fun home() = HomeViewModelController(FetchQuoteUseCaseImpl(QuoteRepositoryImpl()), i18N)`

Reply:
Personally, I do not have a strong opinion for using a `val` vs a `fun`. Would there be any performance advantage of one vs the other ? 

But both prefixing with `create`(1) and or suffixing with `ViewModelController`(2) seems superfluous. 

1. When using the factory pattern, it is a given that it will be "creating" objects. Also, wether it creates one each time, returns always the same instance or anything in between should be an implementation detail of the factory.

2. This is a *ViewModelController* factory, so obviously all methods return *ViewModelControllers*

Here is what the kotlin coding convention has to say about function naming https://kotlinlang.org/docs/reference/coding-conventions.html#function-names TLDR ? : does not help us much here

Reply:
For a factory a function makes more sense IMHO. As a consumer, I would expect a val to return exactly the same thing when called multiple times. This would not be what I expect from a create() method of a factory.

Reply:
+1 for removing the ViewModelController part of the method name.
Or maybe `createHome()` ?